python/nano/src/bigdl/nano/tf/optimizers/test_optimizers.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import numpy as np
+from sparse_adam import SparseAdam
+
+
+def test_optimizer_sparseadam():
+
+    from bigdl.nano.tf.keras.layers import Embedding
+    from bigdl.nano.tf.keras import Sequential
+    import tensorflow as tf
+    model = Sequential()
+    model.add(Embedding(1000, 64, input_length=10, regularizer=tf.keras.regularizers.L2()))
+    sparse_optim = SparseAdam()
+    model.compile(sparse_optim, 'mse')
+    input_array = np.random.randint(1000, size=(32, 10))

```

Comment:
I recommend you explicitly construct the input array. Let's the input length is 5, and  batch_size is 2. You can construct  you input as the following
```
np.array([[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]])
```
then after training one batch, only the first 10 rows of the embedding matrix should be changed.

You can use `train_on_batch` (see https://www.tensorflow.org/api_docs/python/tf/keras/Model#train_on_batch) to train one batch.
You can use `get_weights` (see https://www.tensorflow.org/api_docs/python/tf/keras/layers/Layer#get_weights) before and after `train_on_batch` and compare the results to see if the weights change as expected.

Reply:
Alright. I' try it.